policyengine_us/variables/gov/states/vt/tax/income/adjusted_gross_income/subtractions/retirement_income_exemption/vt_retirement_income_exemption.py

```diff
@@ -0,0 +1,82 @@
+from policyengine_us.model_api import *
+
+
+class vt_retirement_income_exemption(Variable):
+    value_type = float
+    entity = TaxUnit
+    definition_period = YEAR
+    label = "Vermont retirement income exemption"
+    reference = (
+        "https://legislature.vermont.gov/statutes/section/32/151/05811",  # Titl. 32 V.S.A. § 5811(21)(B)(iv)
+        "https://tax.vermont.gov/sites/tax/files/documents/IN-112%20Instr-2022.pdf#page=3",  # Instruction for 2022 SCHEDULE IN-112 - RETIREMENT INCOME EXEMPTION WORKSHEET
+    )
+    unit = USD
+    defined_for = "vt_retirement_income_exemption_eligible"
+    documentation = "Vermont retirement benefits exempt from Vermont taxation."
+
+    def formula(tax_unit, period, parameters):
+        # Filer can choose from one of Social Security, Civil Service Retirement System (CSRS), Military Retirement Income
+        # or other eligible retirement systems to calculate retirement income exemption
+        p = parameters(
+            period
+        ).gov.states.vt.tax.income.agi.retirement_income_exemption
+        # Get social security amount
+        tax_unit_taxable_social_security = tax_unit(
+            "tax_unit_taxable_social_security", period
+        )
+        # Get vt retirement income exclusion from military retirement system
+        vt_military_retirement_pay_exclusion = tax_unit(
+            "vt_military_retirement_pay_exclusion", period
+        )
+        # Get vt retirement income exclusion from CSRS
+        vt_csrs_retirement_pay_exclusion = tax_unit(
+            "vt_csrs_retirement_pay_exclusion", period
+        )
+        # Get vt retirement income exclusion from other certain retirement systems
+        vt_other_retirement_pay_exclusion = tax_unit(
+            "vt_other_retirement_pay_exclusion", period
+        )
+
+        # Assume that filers will always choose the largest reitrement income exclusion from various retirement system
+        # Pair-wise comparsion
+        income_array = [
```

Comment:
does not filer not get to deduct each of those individually? It seems like that at the first glance of the form

Reply:
I think it should be selected from one of the retirement income. The law/form specify that only one should be selected to calculate exemption. 
From https://tax.vermont.gov/individuals/seniors-and-retirees: "Note: A taxpayer can only claim one of the following exemptions, even if more than one applies: Social Security Exemption; Military Retirement Exemption; Civil Service Retirement System Exemption; Other Retirement System Exemption."
From legal code Title 32 V.S.A. § 5830e (e) also reflect this  "Requirement to elect one exclusion....."
https://legislature.vermont.gov/statutes/section/32/151/05830e